src/backend/app.js

```diff
@@ -20,11 +22,73 @@ app.use(express.json());
 app.use(cors());
 
 router.use("/meals", mealsRouter);
-
+// Route to get future meals sorted by ID
+app.get("/future-meals", async (req, res) => {
+  try {
+    const meals = await knex
+      .select()
+      .from("meal")
+      .where("when", ">", knex.fn.now());
+    res.status(200).json(meals);
+  } catch (error) {
+    res.status(500).json("500. Internal Server Error");
+  }
+});
+// Respond with all meals in the past (relative to the when datetime)
+app.get("/past-meals", async (req, res) => {
+  try {
+    const meals = await knex
+      .select()
+      .from("meal")
+      .where("when", "<", knex.fn.now())
+      .orderBy("id");
+    res.status(200).json(meals);
+  } catch (error) {
+    res.status(500).json("500. Internal Server Error");
+  }
+});
+// Respond with all meals sorted by ID
+app.get("/all-meals", async (req, res) => {
+  try {
+    const meals = await knex
+      .select("title", "description")
+      .from("meal")
+      .orderBy("id");
+    res.status(200).json(meals);
+  } catch (error) {
+    res.status(500).json("500. Internal Server Error");
+  }
+});
+//Respond with the first meal (meaning with the minimum id)
+app.get("/first-meal", async (req, res) => {
+  try {
+    const meal = await knex
+      .select("title", "description")
+      .from("meal")
+      .orderBy("id")
+      .first();
+    res.status(200).json(meal);
```

Comment:
Overall looks great. I like how you managed to use the knex library to create your queries, you did a very good job. I just have some observations. 

1. for /first-meal and /last-meal you should return a single object, not sure if you are returning that here.
2. you should return a 404 if this queries don't return any data for /first-meal and /last-meal